sunbeam-python/sunbeam/provider/maas/steps.py

```diff
@@ -2104,10 +2105,17 @@ class MaasClusterStatusStep(ClusterStatusStep):
 
     def models(self) -> list[str]:
         """List of models to query status from."""
-        return [
-            self.deployment.infra_model,
-            self.deployment.openstack_machines_model,
-        ]
+        models = [self.deployment.infra_model]
+        try:
+            run_sync(self.jhelper.get_model(self.deployment.openstack_machines_model))
```

Comment:
Can you use self.jhelper.model_exists(self.deployment.openstack_machines_model) which returns boolean. This is to ensure connection to the model is disconnected once the model is read.